### PR TITLE
[MM-34203] Send team id on bindings call

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The team id was previously not being sent on binding calls. We discussed in our team meeting that we want to send this from the client.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-34203

```release-note
NONE
```

### CODE REVIEW

File: app/mm-redux/actions/apps.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 import {bindClientFunc} from './helpers';
 
 export function fetchAppBindings(userID: string, channelID: string): ActionFunc {
-    return bindClientFunc({
-        clientFunc: () => Client4.getAppsBindings(userID, channelID),
-        onSuccess: AppsTypes.RECEIVED_APP_BINDINGS,
-    });
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const channel = getChannel(getState(), channelID);
+        const teamID = channel?.team_id || '';
```
Comment (larkox) [inline app/mm-redux/actions/apps.ts:15]: Same comment as on webapp. Does it make sense to send the current team id?

**Review by larkox — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Regression tested 2 different
- Tested across various teams
- No regressions found found
- Ensured Team ID is now included 
LGTM!

**Comment by DHaussermann** [comment]
@amyblais would we be able to cherry-pick this and the corresponding webapp https://github.com/mattermost/mattermost-webapp/pull/7902 ?

cc @catalintomai

**Review by catalintomai — APPROVED** [review]

**Comment by amyblais** [comment]
> @amyblais would we be able to cherry-pick this (once dev review is done) and the corresponding webapp [mattermost/mattermost-webapp#7902](https://github.com/mattermost/mattermost-webapp/pull/7902) ?
> 
> cc @catalintomai

@mickmister @jfrerich Please help prioritize this PR.

**Pull request merged**